aws/request/timeout_read_closer_test.go

```diff
@@ -0,0 +1,71 @@
+package request
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"testing"
+	"time"
+)
+
+type testReader struct {
+	duration time.Duration
+}
+
+func (r *testReader) Read(b []byte) (int, error) {
+	time.Sleep(r.duration)
+	return 0, io.EOF
+}
+
+func (r *testReader) Close() error {
+	return nil
+}
+
+func TestTimeoutReadCloser(t *testing.T) {
+	reader := timeoutReadCloser{
+		reader: &testReader{
+			duration: time.Second,
+		},
+		duration: time.Millisecond,
+	}
+	b := make([]byte, 100)
+	_, err := reader.Read(b)
+	if err != nil {
+		t.Log(err)
+	}
+}
+
+func TestTimeoutReadCloserSameDuration(t *testing.T) {
+	reader := timeoutReadCloser{
+		reader: &testReader{
+			duration: time.Millisecond,
+		},
+		duration: time.Millisecond,
+	}
+	b := make([]byte, 100)
+	_, err := reader.Read(b)
+	if err != nil {
+		t.Log(err)
+	}
+}
+
+func TestWithResponseReadTimeout(t *testing.T) {
+	r := Request{
+		HTTPResponse: &http.Response{
+			Body: ioutil.NopCloser(bytes.NewReader(nil)),
+		},
+	}
+	r.ApplyOptions(WithResponseReadTimeout(time.Second))
+	err := r.Send()
+	if err != nil {
+		t.Error(err)
+	}
+	v, ok := r.HTTPResponse.Body.(*timeoutReadCloser)
```

Comment:
With the addition of the `adaptError` function should add test that verifies that functionality as well.